tests/integration/test_interactive.py

```diff
@@ -196,7 +195,10 @@ def test_interactive_publishing_report_success(
         string=re.compile(".*Created by.*")
     )
     spans = [s.parent for s in strings]
-    span = first(spans, key=lambda span: "on" in span.text)
+    try:
+        span = next(sp for sp in spans if "on" in sp.text)  # pragma: no branch
+    except StopIteration:  # pragma: no cover
+        pytest.fail("This test currently asserts a span will be found")
```

Comment:
thought: Previously `first()` would default to `None` as a value if no match is found.

For the tests, I've favoured explicitly adding failure states where we shouldn't ever fail.

This then needs:

* `no branch` because `coverage` complains about [partial branch coverage](https://github.com/nedbat/coveragepy/blob/2a7d9779add1e8fa3425a784347316f5d28a510d/doc/branch.rst#L120-L131); in these cases, the branch where the generator is exhausted is not reached in these cases, as we do find a match
* `no cover` because we should never get the failure state occurring if the tests pass

Reply:
Have removed the commit with these changes.